File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        if len(nums) == 0:
            return[[]]
        
        permList:List[List[int]] = self.permute(nums[1:])
        
        res:List[List[int]] = []
        for sPerm in permList:
            for i in range(len(sPerm)+1):
                permCopy = sPerm.copy()
                permCopy.insert(i,nums[0])
                res.append(permCopy)
        return res
            
                
                
            

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.go
func containsNearbyDuplicate(nums []int, k int) bool {
	abs:=func(num int) int{
		if num<0{
			return num*-1
		}
		return num
	}

	for i,_:=range nums{
		for j := i; j<len(nums);j++{
			if i==j || nums[i] != nums[j]||abs(i-j)>k{
				continue
			}
			return true
		}
	}
	return false

}
